subsystems/mvdm/ntvdm/dos/dos32krnl/dos.c

```diff
@@ -950,8 +965,19 @@ VOID WINAPI DosInt21h(LPWORD Stack)
         case 0x3D:
         {
             WORD FileHandle;
+            BYTE AccessShareModes = getAL();
             LPCSTR FileName = (LPCSTR)SEG_OFF_TO_PTR(getDS(), getDX());
-            WORD ErrorCode = DosOpenFile(&FileHandle, FileName, getAL());
+            WORD ErrorCode = DosOpenFile(&FileHandle, FileName, AccessShareModes);
+
+            /*
+             * Check if we failed because we attempted to open a file for write
+             * on a CDROM drive. In that situation, attempt to reopen for read
```

Comment:
Looks good for commit!

OK I will do my nickpick: missing full stop at the end of the sentence :)